api/src/test/resources/io/strimzi/api/kafka/model/KafkaConnector.out.yaml

```diff
@@ -10,4 +10,19 @@ spec:
   tasksMax: 2
   config:
     file: "/home/source/test.source.txt"
-    topic: "test.topic"
\ No newline at end of file
+    topic: "test.topic"
+status:
+  observedGeneration: 0
+  connector:
```

Comment:
I wanted to have a single property in which to put the JSON connector status which we receive from the API. I'm not attached to the name (`connectorStatus` might be better?). The alternative would be to merge the properties of the result from the API with our own properties.

```
status:
    observedGeneration: ...#ours
    conditions: ...#ours
    connector: # connect's
      state: PAUSED
      worker_id: 172.17.0.11:8083
    name: my-source-connector # connect's
    tasks: # connect's
    - id: 0
      state: RUNNING
      worker_id: 172.17.0.11:8083
    type: source # connect's
```

which would be less clear I think, and risk (though it's not likely) a collision in the future if proerties were added to the connect API response.

Reply:
Tbh using `connectorStatus` under `status` which is about a connector sounds weird to me.
My view is that we are providing a Kubernetes "view" of the connector so mixing information from us and from Connect REST API should be the way to go without trying to define "borders"?. I can get your concern about future collisions but maybe this won't happen :-)

Reply:
Ok, lets keep it as `connector`.

Reply:
Can we think of a better name? I don't like `connector.connector` either. It's `connectorStatus` right now, which I personally think is _slightly_ better, but maybe `apiStatus` is clearer?

Reply:
@tombentley yes I would use something like `api<SOMETHING>`. The only thing I am not sure is using "status" because we already are under the `status` field. Maybe `apiConnect`? Otherwise, let's go for `apiStatus` I think it's better than the current `connectorStatus`.

Reply:
`apiStatus` would suggest a status of some API to me, not a status of some entity inside this API.

Reply:
True. And it's not like where is another source for the status information. So let's settle on `connectorStatus`.